pages/docs/couple-your-code/couple-your-code-porting-v2-3.md

```diff
@@ -26,16 +26,18 @@ Please add breaking changes here when merged to the `develop` branch.
-- Remove `initializeData()` and initialize the data after defining the mesh and before calling `initialize()` if `requiresInitialData()` is `true`.
+- Remove `initializeData()`. The functions `initializeData()` and `ìnitialize()`have been grouped in the new function `initialize()`. Before calling `ìnitialize()` you have to initialize the mesh and the data ( if `requiresInitialData()` is `true`).
 - Remove `isReadDataAvailable()` and `isWriteDataRequired()`, or replace them with your own logic if you are subcycling in your adapter.
 - Remove `getMeshVertices()` and `getMeshVertexIDsFromPositions()`. This information is already known by the adapter.
-- Replace `precice::constants::*` with `isActionRequired()` and `markActionFulfilled()` with their respective requirement clause: `requiresInitialData()`, `requiresReadingCheckpoint()` or `requiresWritingCheckpoint()`. If these requirements are checked, then they are promised to be acted on.
+- Replace `isActionRequired()` with their respective requirement clause: `requiresInitialData()`, `requiresReadingCheckpoint()` or `requiresWritingCheckpoint()`.
+- Remove `markActionFullfiled()`. If `requiresInitialData()`, `requiresReadingCheckpoint()` or `requiresWritingCheckpoint()` are called, then they are promised to be acted on so `markActionFullfiled()` is no longer needed.
 - Replace `isMeshConnectivityRequired` with `requiresMeshConnectivityFor`
 - Replace `isGradientDataRequired` with `requiresGradientDataFor`
 - Remove the now obsolete calls to `getMeshID()` and `getDataID()`.
 - Remove `hasMesh()` and `hasData()`.
 - Change integer input argument mesh ID to a string with the mesh name in the API commands `hasMesh`, `requiresMeshConnectivityFor`, `setMeshVertex`, `getMeshVertexSize`, `setMeshVertices`, `setMeshEdge`, `setMeshEdges`, `setMeshTriangle`, `setMeshTriangles`, `setMeshQuad`, `setMeshQuads`, `setMeshTetrahedron`, `setMeshTetrahedrons`, `setMeshAccessRegion`, `getMeshVerticesAndIDs`.
-- Change integer input argument data ID to string arguments mesh name and data name in the API commands `hasData`, `writeBlockVectorData`, `writeVectorData`, `writeBlockScalarData`, `writeScalarData`, `readBlockVectorData`, `readVectorData`, `readBlockScalarData`, `readScalarData`, `requiresGradientDataFor`, `writeBlockVectorGradientData`, `writeVectorGradientData`, `writeBlockScalarGradientData`, `writeScalarGradientData`.
+- Replace `writeBlockVectorData`, `writeVectorData`, `writeBlockScalarData`, `writeScalarData`, `readBlockVectorData`, `readVectorData`, `readBlockScalarData`, `readScalarData`, `requiresGradientDataFor`, `writeBlockVectorGradientData`, `writeVectorGradientData`, `writeBlockScalarGradientData` and `writeScalarGradientData` by the new general functions `readData`, `writeData` and `writeGradientData`.
+- Change integer input argument data ID to string arguments mesh name and data name in the API commands `hasData`, `readData`, `writeData` and `writeGradientData`.
```

Comment:
An example here would be helpful. But we should rather (in a second phase) create subsections with detailed examples and hints.

Reply:
Looking at the other examples that already exist I guess this particular example could look like 

```diff
- couplingInterface.writeBlockVectorData(displacementID displacementSize, displacementIndices, displacementValues);
+ couplingInterface.writeData(displacementName, displacementSize, displacementIndices, displacementValues);
```

But as you say  I guess this can wait for the second phase as you say. 

Also a comment to the existing examples: the  porting guide says
>    Consider renaming your objects from, e.g., `interface` to `participant`, to better reflect the purpose and to be consistent with the rest of the changes.

However the existing examples use an object named `couplingInterface` instead of e.g. `preciceParticipant`:


```diff
- couplingInterface.writeBlockVectorData(displacementID displacementSize, displacementIndices, displacementValues);
+ preciceParticipant.writeData(displacementName, displacementSize, displacementIndices, displacementValues);
```

Reply:
> But as you say I guess this can wait for the second phase as you say.

Yes, let's keep it for a separate PR. Many things need more explanation / examples.

> However the existing examples use an object named couplingInterface instead of e.g. preciceParticipant:

This is also something we need to update in the https://github.com/precice/precice.github.io/tree/precice-v3 branch.